src/analysis/retail/warlock/demonology/modules/features/SummonDemonicTyrant.tsx

```diff
@@ -1,35 +1,78 @@
 import SPELLS from 'common/SPELLS';
 import TALENTS from 'common/TALENTS/warlock';
 import Analyzer, { Options, SELECTED_PLAYER, SELECTED_PLAYER_PET } from 'parser/core/Analyzer';
-import Events, { ApplyBuffEvent } from 'parser/core/Events';
+import Events, {
+  AnyEvent,
+  ApplyBuffEvent,
+  BeginCastEvent,
+  BeginChannelEvent,
+  EndChannelEvent,
+  GlobalCooldownEvent,
+} from 'parser/core/Events';
 import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
 import Statistic from 'parser/ui/Statistic';
 import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
-
 import DemoPets from '../pets/DemoPets';
 import './SummonTyrant.scss';
+import SpellLink from 'interface/SpellLink';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+import ContextualSpellUsageSubSection from 'parser/core/SpellUsage/HideGoodCastsSpellUsageSubSection';
+import { ChecklistUsageInfo, SpellUse } from 'parser/core/SpellUsage/core';
+import { CastEvent } from 'parser/core/Events';
+import { combineQualitativePerformances } from 'common/combineQualitativePerformances';
+import Casts from 'interface/report/Results/Timeline/Casts';
+import EmbeddedTimelineContainer, {
+  SpellTimeline,
+} from 'interface/report/Results/Timeline/EmbeddedTimeline';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
 
 const debug = false;
 
+const CASTS_TO_REMOVE_FROM_TIMELINE = [
+  TALENTS.CALL_DREADSTALKERS_TALENT.id,
+  TALENTS.SUMMON_DEMONIC_TYRANT_TALENT.id,
+  SPELLS.HAND_OF_GULDAN_CAST.id,
+  SPELLS.DEMONBOLT.id,
+  SPELLS.SHADOW_BOLT_DEMO.id,
+  TALENTS.DEMONIC_GATEWAY_TALENT.id,
+  SPELLS.INQUISITORS_GAZE_CAST.id,
+];
+
+interface TyrantCast {
+  event: CastEvent;
+  castNumber: number;
+  gfgOnCd: boolean;
+}
+
 class SummonDemonicTyrant extends Analyzer {
   static dependencies = {
     demoPets: DemoPets,
+    spellUsable: SpellUsable,
   };
 
   demoPets!: DemoPets;
+  spellUsable!: SpellUsable;
   _reignOfTyrannyPower!: number[];
   _hasReignOfTyranny!: boolean;
-  _petsPerCast!: Record<number, number>[];
+  _hasGFG!: boolean;
 
   _summsWithDemonicPower!: { [id: string]: any }[];
-  _tyrantCast!: number;
+  _tyrantsCast!: number;
```

Comment:
I would suggest assigning a value to this field (and the others) instead of using the `!` syntax; you can override it in the constructor to be the actual value if so desired.